Added option to exempt tamed hostile mobs from restrictions on carrying hostile mobs.

KandiPanda opened the pull request:
What the title says. One example use case would be for the Foxhound mobs from [the Quark mod](http://vazkii.net/subdom/quark/old/), which remain marked as hostile even when tamed, to allow carrying them _when tamed_ without allowing any other (untamed) hostile mobs to be carried.

Resolves #218

KandiPanda committed:
Update Configs.java

Add tamedHostileMobExemption option

KandiPanda committed:
Update PickupHandler.java

NotStirred reviewed (changes requested):
In PickupHandler
I suggest removing the enclosed `if (toPickUp instanceof EntityTameable)` on line 234.
Possibly also remove the redundant if on 227

KandiPanda commented:
> Possibly also remove the redundant if on 227

That (and the matching one on line 295) were in the code before I did anything with it, I assumed that it was there for a reason and didn't mess with it.

NotStirred commented on src/main/java/tschipp/carryon/common/handler/PickupHandler.java:
```diff
@@ -219,73 +219,146 @@ public static boolean canPlayerPickUpEntity(EntityPlayer player, Entity toPickUp
-									String condition = CustomPickupOverrideHandler.getPickupCondition(toPickUp);
-									boolean has = (boolean) hasStage.invoke(stageData, condition);
-
-									if (has)
-										return handleProtections((EntityPlayerMP) player, toPickUp);
+									EntityTameable tame = (EntityTameable) toPickUp;
+									UUID owner = tame.getOwnerId();
+									UUID playerID = player.getUUID(player.getGameProfile());
+									if (owner != null && !owner.equals(playerID))
+										return false;
```
This entire if is redundant due to the check for the same thing above.

NotStirred commented on src/main/java/tschipp/carryon/common/handler/PickupHandler.java:
```diff
@@ -219,73 +219,146 @@ public static boolean canPlayerPickUpEntity(EntityPlayer player, Entity toPickUp
-									return false;
+								if (toPickUp instanceof EntityTameable)
+								{
+									EntityTameable tame = (EntityTameable) toPickUp;
+									UUID owner = tame.getOwnerId();
```
```suggestion
									UUID owner = tamecheck.getOwnerId();
```

NotStirred reviewed (changes requested):
Added some revisions

NotStirred commented on src/main/java/tschipp/carryon/common/handler/PickupHandler.java:
```diff
@@ -219,73 +219,146 @@ public static boolean canPlayerPickUpEntity(EntityPlayer player, Entity toPickUp
-								UUID playerID = player.getUUID(player.getGameProfile());
-								if (owner != null && !owner.equals(playerID))
-									return false;
+								if (toPickUp instanceof EntityTameable)
+								{
+									EntityTameable tame = (EntityTameable) toPickUp;
+									UUID owner = tame.getOwnerId();
+									UUID playerID = player.getUUID(player.getGameProfile());
+									if (owner != null && !owner.equals(playerID))
+										return false;
+								}
```
```suggestion
								UUID owner = tamecheck.getOwnerId();
								UUID playerID = player.getUUID(player.getGameProfile());
								if (owner != null && !owner.equals(playerID))
									return false;
```

KandiPanda committed:
Update PickupHandler.java

Removed redundant EntityTameable check from within code already covered by another EntityTameable check (would always be "true" anyways).
Removed redundant EntityTameable check and contents from within the "else" of same outer check (else means it's already failed, and will never succeed this check or run the code within it anyways).

Tschipp commented:
Thanks, will take a look soon.

Tschipp commented:
Looks good! Thank you for making this!
